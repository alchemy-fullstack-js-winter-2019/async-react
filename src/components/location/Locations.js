import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { withPaging } from '../paging/Paging';
import { getLocations } from '../../services/rickAndMortyApi';
import Location from './Location';
import { withFetch } from '../fetch/Fetch';

export class Locations extends PureComponent {
  static propTypes = {
    results: PropTypes.array.isRequired
  };
  
  render() {
    const locationRows = this.props.results.map(location => {
      return <li className="locations" key={location.id}><Location location={location}/></li>;
    });
    return (
      <ul>
        {locationRows}
      </ul>
    );
  }
}
const FetchLocation = withFetch(Locations)(getLocations);

export const LocationsWithPaging = withPaging(FetchLocation);
