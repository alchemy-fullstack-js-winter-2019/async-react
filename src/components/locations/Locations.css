.locations {
  display: grid;
  grid-template-columns: repeat(3, 321px);
  grid-template-rows: repeat(10, 321px);
  grid-gap: 10px;
  flex-direction: row;
  flex-wrap: wrap;
}