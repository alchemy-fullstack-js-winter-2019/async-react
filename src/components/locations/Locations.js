import Location from './Location';
import Results from '../paging/Results';
import { getLocations } from '../../services/rickAndMortyApi';

export default Results(Location, getLocations);
