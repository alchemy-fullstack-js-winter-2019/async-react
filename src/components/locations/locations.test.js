import React from 'React';
import renderer from 'react-test-renderer';
import Locations from './Locations';

jest.mock('../../services/rickAndMortyApi.js');

describe('Characters', () => {
  it('matches a snapshot', () => {
    const tree = renderer.create(<Locations />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
