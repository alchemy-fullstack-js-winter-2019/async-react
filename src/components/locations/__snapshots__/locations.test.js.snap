// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters matches a snapshot 1`] = `
Array [
  <h2
    className="subTitle"
  />,
  <div
    className="buttons"
  >
    <button
      onClick={[Function]}
    >
      Back
    </button>
    <p
      className="pageInfo"
    >
      1
       / 
      1
    </p>
    <button
      onClick={[Function]}
    >
      Forward
    </button>
  </div>,
  <div>
    <h3>
      Locations
    </h3>
    <ul />
  </div>,
]
`;
